cli/cmdlineparser.cpp

```diff
@@ -1227,6 +1227,28 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
                 mSettings.reportProgress = tmp;
             }
 
+            else if (std::strncmp(argv[i], "--report-type=", 14) == 0) {
+                const std::string typeStr = argv[i] + 14;
+                if (typeStr == "normal") {
+                    mSettings.reportType = ReportType::normal;
+                } else if (typeStr == "autosar") {
+                    mSettings.reportType = ReportType::autosar;
+                } else if (typeStr == "certC") {
```

Comment:
I think it would be preferable to match the `--premium` options here. So the options should be:

     autosar
     cert-c-2016
     cert-cpp-2016
     misra-c-2012
     misra-c-2023
     misra-cpp-2008
     misra-cpp-2023

internally I think the enum constants can be named as-is. We can easily rename the internal enum constants without breaking anything later.